src/celeritas/neutron/interactor/detail/InvariantQ2Sampler.hh

```diff
@@ -0,0 +1,363 @@
+    real_type max_Q2_;
+    // Parameters for the CHIPS differential cross section
+    ExchangeParameters par_Q2_;
+
+    //// HELPER FUNCTIONS ////
+
+    // Sample the slope
+    template<class Engine>
+    inline CELER_FUNCTION real_type sample_q2(real_type radius, Engine& rng)
+    {
+        return -std::log(1 - radius * generate_canonical(rng));
+    }
+
+    // Calculate the maximum momentum transfer
+    inline CELER_FUNCTION real_type calc_max_Q2(real_type p) const;
+
+    // Calculate parameters used in the quark-exchange process
+    inline CELER_FUNCTION ExchangeParameters calc_par_Q2(real_type p) const;
+
+    //// COMMON PROPERTIES ////
+
+    // Covert from clhep::MeV value to clhep::GeV value
+    static CELER_CONSTEXPR_FUNCTION real_type to_gev() { return 1e-3; }
+
+    // S-wave for neutron p < 14 MeV/c (kinetic energy < 0.1 MeV)
+    static CELER_CONSTEXPR_FUNCTION real_type s_wave_limit() { return 1.4e-2; }
+};
+
+//---------------------------------------------------------------------------//
+// Inline DEFINITIONS
+//---------------------------------------------------------------------------//
+/*!
+ * Construct from the incident momentum in the lab frame and the target(A).
+ *
+ * \note The incident neutron momentum, and neutron and nucleus masses are
+ *  converted to the GeV value.
+ */
+CELER_FUNCTION
+InvariantQ2Sampler::InvariantQ2Sampler(NeutronElasticRef const& shared,
+                                       IsotopeView const& target,
+                                       real_type neutron_p)
+
+    : par_(shared.Q2.coeffs[target.isotope_id()].par)
+    , neutron_mass_(value_as<Mass>(shared.neutron_mass) * this->to_gev())
+    , target_mass_(value_as<Mass>(target.nuclear_mass()) * this->to_gev())
+    , A_(static_cast<real_type>(target.atomic_mass_number().get()))
+    , heavy_target_(A_ > 6)
+    , neutron_p_(neutron_p * this->to_gev())
+{
+    max_Q2_ = calc_max_Q2(neutron_p_);
+    par_Q2_ = calc_par_Q2(neutron_p_);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Sample the momentum transfer of the neutron-nucleus elastic scattering
+ * based on G4ChipsElasticModel and G4ChipsNeutronElasticXS of the Geant4
+ * 11.2 release.
+ *
+ */
+template<class Engine>
+CELER_FUNCTION auto InvariantQ2Sampler::operator()(Engine& rng) -> real_type
+{
+    // Sample \f$ Q^{2} \f$ below S-wave limit
+    if (neutron_p_ < this->s_wave_limit())
+    {
+        return max_Q2_ * generate_canonical(rng) / ipow<2>(this->to_gev());
+    }
+
+    // Sample \f$ Q^{2} \f$
+    real_type q2 = 0;
+
+    if (A_ == 1)
+    {
+        // Special case for the \f$ n + p \rightarrow n + p \f$ channel
+        real_type R1 = -std::expm1(-max_Q2_ * par_Q2_.slope[0]);
+        real_type R2 = -std::expm1(-max_Q2_ * par_Q2_.slope[1]);
+
+        real_type I1 = R1 * par_Q2_.expnt[0];
+        real_type I2 = R2 * par_Q2_.expnt[1] / par_Q2_.slope[1];
```

Comment:
No capital variable names. And you could even write this as
```c++
real_type const r[] = {..., ...};
```
and use `r[0]` and `r[1]`.